pkg/goanalysis/runners.go

```diff
@@ -92,11 +94,34 @@ func buildIssues(diags []Diagnostic, linterNameBuilder func(diag *Diagnostic) st
 			text = fmt.Sprintf("%s: %s", diag.Analyzer.Name, diag.Message)
 		}
 
+		var suggestedFixes []analysis.SuggestedFix
+
+		for _, sf := range diag.SuggestedFixes {
+			nsf := analysis.SuggestedFix{Message: sf.Message}
+
+			for _, edit := range sf.TextEdits {
+				end := edit.End
+
+				if !end.IsValid() {
+					end = edit.Pos
+				}
+
+				nsf.TextEdits = append(nsf.TextEdits, analysis.TextEdit{
+					Pos:     token.Pos(diag.File.Offset(edit.Pos)),
+					End:     token.Pos(diag.File.Offset(end)),
+					NewText: edit.NewText,
+				})
```

Comment:
What's going on here? Why do we need to convert the suggested fix from the analyzer and create a new one with this offset? Can we document briefly why that is?

Reply:
I added a comment.